src/main/java/mate/academy/spring/service/impl/UserServiceImp.java

```diff
@@ -15,14 +18,52 @@
 
     @Autowired
     private UserDao userDao;
+    @Autowired
+    private RoleService roleService;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
 
     @Transactional
     @Override
-    public User add(User user) {
+    public User add(UserRegistrationDto userDto) throws IllegalArgumentException {
+        if (emailExist(userDto.getEmail())) {
+            throw new IllegalArgumentException(
+                    "There is an account with that email adress: "
+                            + userDto.getEmail());
+        }
+        if (usernameExist(userDto.getUserName())) {
+            throw new IllegalArgumentException(
+                    "There is an account with that username: "
+                            + userDto.getUserName());
+        }
+        User user = new User();
+        user.setName(userDto.getFirstName());
```

Comment:
better create separate method for mapping dto to entity in DtoUtil class

Reply:
done